src/main/java/com/microsoft/azure/kusto/kafka/connect/sink/FileWriter.java

```diff
@@ -94,26 +94,44 @@ boolean isDirty() {
 
     public void openFile(@Nullable Long offset) throws IOException {
         SourceFile fileProps = new SourceFile();
-
         File folder = new File(basePath);
         if (!folder.exists() && !folder.mkdirs()) {
             if (!folder.exists()) {
                 throw new IOException(String.format("Failed to create new directory %s", folder.getPath()));
             }
             log.warn("Couldn't create the directory because it already exists (likely a race condition)");
         }
-
         String filePath = getFilePath.apply(offset);
         fileProps.path = filePath;
         // Sanitize the file name just be sure and make sure it has the R/W permissions only
-        File file = new File(FilenameUtils.normalize(filePath));
-        boolean execResult = file.setExecutable(false);
-        execResult = execResult && file.setReadable(true);
-        execResult = execResult && file.setWritable(true);
-        execResult = execResult && file.createNewFile();
-        if (!execResult) {
-            log.warn("Setting permissions creating file {} returned false.", filePath);
+        String sanitizedFilePath = FilenameUtils.normalize(filePath);
+        if (sanitizedFilePath == null) {

```

Comment:
I don't know if its a real use case but this breaks users that sets temp dir to a relative path that uses dots.. 
i.e "tempdir.path" : "../tempdir" ?
Maybe we can check in setup if the tempdir is relative -then canonicalize it or just canonicalize it here instead of normalize.